src/payment/strategies/barclaycard/barclaycard-payment-strategy.spec.ts

```diff
@@ -0,0 +1,194 @@
+
+        jest.spyOn(orderActionCreator, 'finalizeOrder')
+            .mockReturnValue(finalizeOrderAction);
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        jest.spyOn(paymentActionCreator, 'initializeOffsitePayment')
+            .mockReturnValue(initializeOffsitePaymentAction);
+
+        strategy = new BarclaycardPaymentStrategy(store, orderActionCreator, paymentActionCreator);
+    });
+
+    it('submits order with payment payload', async () => {
+        await strategy.execute(payload, options);
+
+        expect(orderActionCreator.submitOrder).toHaveBeenCalledWith(payload, options);
+        expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+    });
+
+    it('initializes offsite payment flow', async () => {
+        const expectedPayload = {
+            formattedPayload: {
+                browserInfo: {
+                    deviceOs: expect.any(String),
+                    language: expect.any(String),
+                    screenResolution: expect.any(String),
+                    timeZone: expect.any(String),
+                },
+            },
+        };
+        await strategy.execute(omit(payload, 'payment.paymentData'), options);
+
+        expect(paymentActionCreator.initializeOffsitePayment).toHaveBeenCalledWith(options.methodId, options.gatewayId, expect.objectContaining(expectedPayload));
+        expect(store.dispatch).toHaveBeenCalledWith(initializeOffsitePaymentAction);
+    });
+
+    it('finalizes order if order is created and payment is acknowledged', async () => {
+        const state = store.getState();
+
+        jest.spyOn(state.order, 'getOrder')
+            .mockReturnValue(getOrder());
+
+        jest.spyOn(state.payment, 'getPaymentStatus')
+            .mockReturnValue(paymentStatusTypes.ACKNOWLEDGE);
+
+        await strategy.finalize(options);
+
+        expect(orderActionCreator.finalizeOrder).toHaveBeenCalledWith(getOrder().orderId, options);
+        expect(store.dispatch).toHaveBeenCalledWith(finalizeOrderAction);
+    });
+
+    it('finalizes order if order is created and payment is finalized', async () => {
+        const state = store.getState();
+
+        jest.spyOn(state.order, 'getOrder')
+            .mockReturnValue(getOrder());
+
+        jest.spyOn(state.payment, 'getPaymentStatus')
+            .mockReturnValue(paymentStatusTypes.FINALIZE);
+
+        await strategy.finalize(options);
+
+        expect(orderActionCreator.finalizeOrder).toHaveBeenCalledWith(getOrder().orderId, options);
+        expect(store.dispatch).toHaveBeenCalledWith(finalizeOrderAction);
+    });
+
+    it('does not finalize order if order is not created', async () => {
+        const state = store.getState();
+
+        jest.spyOn(state.order, 'getOrder').mockReturnValue(getIncompleteOrder());
+
+        try {
```

Comment:
For exceptions, it's better to test by using `expect(() => ...).toThrow()` rather than try catch, but just a suggestion
```js
expect(() => strategy.finalize()).toThrow(OrderFinalizationNotRequiredError)
```